feature-gate/program/src/instruction.rs

```diff
@@ -44,6 +60,36 @@ impl FeatureGateInstruction {
     }
 }
 
+/// Creates an 'ActivateFeature' instruction.
+pub fn activate_feature(feature_id: &Pubkey) -> Instruction {
+    let accounts = vec![
+        AccountMeta::new(*feature_id, true),
+        AccountMeta::new_readonly(system_program::id(), false),
+    ];
+
+    let data = FeatureGateInstruction::ActivateFeature.pack();
+
+    Instruction {
+        program_id: crate::id(),
+        accounts,
+        data,
+    }
+}
+
+/// Creates a set of two instructions:
+///   * One to fund the feature account with rent-exempt lamports
+///   * Another is the Feature Gate Program's 'ActivateFeature' instruction
+pub fn activate_feature_with_rent_transfer(
+    feature_id: &Pubkey,
+    payer: &Pubkey,
+) -> [Instruction; 2] {
+    let lamports = Rent::default().minimum_balance(Feature::size_of());
+    [
+        system_instruction::transfer(payer, feature_id, lamports),
```

Comment:
I think we should move transfer handling into `process_activate_feature()`. That way we can use the Rent syscall to be sure we're using correct rates. `Rent::default()` may not always be accurate.
Would be something like what we do in spl-assocated-token: https://github.com/solana-labs/solana-program-library/blob/5ce6eace7c128c3eb5686e02de983ee431b73ce1/associated-token-account/program/src/tools/account.rs#L28-L42

Unless there's a reason I'm missing that the transfer needs to be separate?

Reply:
I was following Token2022's example with how account initialization is handled, specifically initializing token metadata:

https://github.com/solana-labs/solana-program-library/blob/acabef3680e972baf8e9a47531c35eaa45405740/token/client/src/token.rs#L3614-L3637

https://github.com/solana-labs/solana-program-library/blob/acabef3680e972baf8e9a47531c35eaa45405740/token/client/src/token.rs#L3657-L3695

But in hindsight I'm thinking that might be the process for some of the Token2022 stuff since the account in question _may_ or _may not_ exist prior to the instruction (?).

Considering these feature accounts will not exist prior to sending the instruction, and our CLI command `solana feature activate` will always use the "with rent" one, I think it makes sense to combine in the program's processor, yep.

Reply:
Note this also requires a payer though

Reply:
I've added a commit for this. If we decide we don't like it I can always pluck it!